src/dynamics/env.py

```diff
@@ -38,68 +39,47 @@ def __init__(
         self.time = self.start_time
         self.reward_fn = reward_fn
 
+    @beartype
     @override
     def reset(
         self,
         *,
         seed: int | None = None,
         options: dict[str, Any] | None = None,
-    ) -> tuple[FloatArray, dict[str, Any]]:
+    ) -> tuple[StateDict, dict[str, Any]]:
         """Reset the state."""
+        super().reset(seed=seed, options=options)
         self.state = self.initial_state
         self.time = self.start_time
-        return super().reset(seed=seed, options=options)
+        return self.state.asdict(), {} if options is None else options
```

Comment:
I added beartype for runtime type checking, and there were two big adjustments this caused, that in practice don't seem to do anything. This is the first, `super().reset(...)` just seeds the function and returns `None`, you're meant to then implement your own reset. I've put this here, but in all honesty I'm not sure if it's ever really used.